apps/site/lib/site_web/controllers/route_controller.ex

```diff
@@ -6,18 +6,42 @@ defmodule SiteWeb.RouteController do
   alias Leaflet.MapData.Polyline
   alias Routes.{Repo, Route}
 
+  @display_fare_classes [
+    :local_bus_fare,
+    :express_bus_fare,
+    :rapid_transit_fare,
+    :commuter_rail_fare,
+    :ferry_fare
+  ]
+
   @spec get_by_stop_id(Plug.Conn.t(), map) :: Plug.Conn.t()
   def get_by_stop_id(conn, %{"stop_id" => stop_id} = _params) do
+    routes = stop_id |> Repo.by_stop()
+
     routesWithPolylines =
-      stop_id
-      |> Repo.by_stop()
+      routes
       |> Enum.map(fn route ->
         route
         |> Route.to_json_safe()
         |> Map.put(:polylines, route_polylines(route, stop_id))
       end)
 
-    json(conn, routesWithPolylines)
+    fares =
+      stop_id
+      # Get fare_class for all routes at this stop and at connecting stops
+      |> Repo.by_stop(include: "stop.connecting_stops")
+      |> Enum.map(&display_fare_class/1)
+      |> Enum.uniq()
+      # Sort in same order as @display_fare_classes
+      |> Enum.sort_by(&Enum.find_index(@display_fare_classes, fn fc -> fc == &1 end))
+      |> Enum.map(&Fares.Repo.for_fare_class(&1, duration: :single_trip, reduced: nil))
+      |> Enum.map(fn fares_for_class ->
+        price = price_range(fares_for_class)
+        name = List.first(fares_for_class) |> fare_name()
+        {name, price}
+      end)
+
+    json(conn, %{routes: routesWithPolylines, fares: fares})
```

Comment:
This is a pattern I am not a huge fan of.  We keep appending new data to existing models instead of making it a new endpoint.  In this case I think it would benefit both flexibility and reuse for making fares its own endpoint to fetch all the fares by stop id, instead of modifying the routes endpoint to return an array of routes and a separate array of fares

Reply:
Have reverted this and made an endpoint for it.